File: src/main/java/com/github/nymann/commitrefactoring/listeners/CommitListener.java
package com.github.nymann.commitrefactoring.listeners;


import com.github.nymann.commitrefactoring.SingletonRefactoringStore;
import com.intellij.notification.Notification;
import com.intellij.notification.Notifications;
import org.jetbrains.annotations.NotNull;

public class CommitListener implements Notifications {
    @Override
    public void notify(@NotNull Notification notification) {
        if (notification.getContent().contains("committed: Refactoring")) {
            SingletonRefactoringStore.getInstance().clear();
        }
    }
}

File: src/main/kotlin/com/github/nymann/commitrefactoring/listeners/CommitListener.kt
package com.github.nymann.commitrefactoring.listeners

import com.github.nymann.commitrefactoring.RefactoringStore
import com.intellij.notification.Notification
import com.intellij.notification.Notifications

class CommitListener : Notifications {
    override fun notify(notification: Notification) {
        if (notification.content.contains("committed: Refactoring")) {
            RefactoringStore.clear()
        }
    }
}